src/Vault.sol

```diff
@@ -20,25 +21,43 @@ contract Vault {
     address public beneficiaryAddress;
 
     // @notice Total amount of ETH deposited to this contract
-    uint256 public totalBalanceEth;
+    uint256 public totalDepositedEth;
 
     // @notice Mapping of user's addresses and amount of ETH they have deposited to this contract (represented as imETH)
     mapping(address => uint256) public userBalance;
 
+    // @notice Boolean variable that indicates if the contract is active or not
+    bool public isContractActive;
+
+    // @notice Event that is emitted when a new user deposits ETH to the contract
+    event Deposit(address indexed user, uint256 amount);
+
+    // @notice Event that is emitted when a user withdraws ETH from the contract
+    event Withdraw(address indexed user, uint256 amount);
+
+    // @notice Event that is emitted when sktakeToLido function is called
+    event StakeToLido(uint256 amount);
+
+    // @notice Event that is emitted when rewardsa re harvested
+    event HarvestRewards(uint256 amount);
+
     constructor(address _stETHaddress, address _beneficiary, address _imETHaddress) {
         stETH = IstETH(_stETHaddress);
         beneficiaryAddress = _beneficiary;
         imETH = ImpactETHtoken(_imETHaddress);
+        isContractActive = true;
     }
 
     /**
         @notice This function allows users to deposit ETH to the contract. The amount of ETH deposited will be minted as imETH tokens
      */
     function deposit() external payable {
+        require(isContractActive, 'Contract is paused');
```

Comment:
this required could be a modifier, something like: **onlyActiveContract**

Reply:
Agreed. Just implemented Pausable instead, please review :)